monai/data/nifti_writer.py

```diff
@@ -9,33 +9,126 @@
+    happen.  otherwise this function will resample `data` using the coordinate
+    transform computed from `affine` and `target_affine`.  Note that the shape
+    of the resampled `data` may subject to some rounding errors. For example,
+    resampling a 20x20 pixel image from pixel size (1.5, 1.5)-mm to (3.0,
+    3.0)-mm space will return a 10x10-pixel image.  However, resampling a
+    20x20-pixel image from pixel size (2.0, 2.0)-mm to (3.0, 3.0)-mma space
+    will output a 14x14-pixel image, where the image shape is rounded from
+    13.333x13.333 pixels. In this case `output_shape` could be specified so
+    that this function writes image data to a designated shape.
+
+    when `affine` and `target_affine` are None, the data will be saved with an
+    identity matrix as the image affine.
+
+    This function assumes the NIfTI dimension notations, that is, the first 3
+    dimensions are spatial dimensions. When saving multiple time steps or
+    multiple channels, time and/or modality axes should be appended after the
+    first three dimensions.
 
-def write_nifti(data, file_name, affine=None, target_affine=None, dtype=np.float32):
-    """Write numpy data into nifti files to disk.
 
     Args:
         data (numpy.ndarray): input data to write to file.
-        affine (numpy.ndarray): affine information for the data.
         file_name (string): expected file name that saved on disk.
-        target_affine (numpy.ndarray, optional):
-            before saving the (data, affine), transform the data into the orientation defined by `target_affine`.
+        affine (numpy.ndarray): the current affine of `data`. Defaults to `np.eye(4)`
+        target_affine (numpy.ndarray, optional): before saving
+            the (`data`, `affine`) as a Nifti1Image,
+            transform the data into the coordinates defined by `target_affine`.
+        resample (bool): whether to run resampling when the target affine
+            could not be achieved by swapping/flipping data axes.
+        output_shape (None or tuple of ints): output image shape.
+            this option is used when resample = True.
+        interp_order (int): the order of the spline interpolation, default is 3.
+            The order has to be in the range 0 - 5.
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.affine_transform.html
+            this option is used when `resample = True`.
+        mode (`reflect|constant|nearest|mirror|wrap`):
+            The mode parameter determines how the input array is extended beyond its boundaries.
+            this option is used when `resample = True`.
+        cval (scalar): Value to fill past edges of input if mode is "constant". Default is 0.0.
+            this option is used when `resample = True`.
         dtype (np.dtype, optional): convert the image to save to this data type.
     """
     assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+    sr = min(data.ndim, 3)
```

Comment:
Why only support <= 3 dims?

Reply:
spatial 3d, because we don't transpose between spatial dimension and channel dimension in this method